src/RefractAPI.cc

```diff
@@ -195,116 +234,125 @@ namespace drafter {
+        element->meta[SerializeKey::Classes] = refract::ArrayElement::Create(metaClass);
 
         if (!contentType.empty()) {
+            // FIXME: has SourceMap?
             element->attributes[SerializeKey::ContentType] = refract::IElement::Create(contentType);
         }
 
         return element;
     }
 
-    refract::IElement* PayloadToRefract(const snowcrash::Payload* payload, const snowcrash::Action* action = NULL)
+    refract::IElement* PayloadToRefract(const SectionInfo<snowcrash::Payload>& payload, const SectionInfo<snowcrash::Action>& action)
     {
         refract::ArrayElement* element = new refract::ArrayElement;
         RefractElements content;
 
         // Use HTTP method to recognize if request or response
-        if (action == NULL || action->method.empty()) {
+        if (action.isNull() || action.section.method.empty()) {
             element->element(SerializeKey::HTTPResponse);
 
-            if (payload != NULL && !payload->name.empty()) {
-                element->attributes[SerializeKey::StatusCode] = refract::IElement::Create(payload->name);
+            if (!payload.isNull()) {
+                element->attributes[SerializeKey::StatusCode] = PrimitiveToRefract(MAKE_SECTION_INFO(payload, name));
             }
         }
         else {
             element->element(SerializeKey::HTTPRequest);
-            element->attributes[SerializeKey::Method] = refract::IElement::Create(action->method);
+            element->attributes[SerializeKey::Method] = PrimitiveToRefract(MAKE_SECTION_INFO(action, method));
 
-            if (payload != NULL) {
-                element->attributes[SerializeKey::Title] = refract::IElement::Create(payload->name);
+            if (!payload.isNull()) {
+                element->attributes[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(payload, name));
             }
         }
 
         // If no payload, return immediately
-        if (payload == NULL) {
+        if (payload.isNull()) {
             element->set(content);
             return element;
         }
 
-        if (!payload->headers.empty()) {
-            element->attributes[SerializeKey::Headers] = CollectionToRefract<refract::ArrayElement>(payload->headers, HeaderToRefract, SerializeKey::HTTPHeaders, refract::IElement::rFull);
+        if (!payload.section.headers.empty()) {
+            element->attributes[SerializeKey::Headers] = CollectionToRefract<refract::ArrayElement>(MAKE_SECTION_INFO(payload, headers),
+                                                                                                    HeaderToRefract,
+                                                                                                    SerializeKey::HTTPHeaders,
+                                                                                                    refract::IElement::rFull);
```

Comment:
Why?  There are much longer lines than this one. eg:
https://github.com/apiaryio/drafter/pull/148/files#diff-2ab1b5da6ee8cb5b29287e4ebfd30e94R243

FYI: linked line was longer than this one, before i made change on it.